PIVX-Project/PIVX
# [TierTwo] ChainLocks (Dip 8)

This PR implements ChainLocks, which is our first real application of Signing Sessions. They are a threshold signature generated by an active quorum that can validate a block among a list of (valid) alternatives, thus preventing forks and explicitly creating a unique valid blockchain.

In particular:
Chainlocks are enabled/disabled with the new spork  SPORK_23_CHAINLOCKS_ENFORCEMENT;
Chainlocks are received and sent with the usual inventory system and with the new message type CLSIG.

For testers first of all:

- Create a local regtest network
- Send SPORK21 and mine enough blocks to activate v6.0
- Create 3 deterministic masternodes
- Wait for a quorum to be successfully created and mined on chain
- Send SPORK23

Then chainlocks should be automatically generated by the active quorum. To check if a given block has a chainlock the rpc getblock has been updated and will show True if the block has a chainlock and False otherwise.
